api/api.go

```diff
@@ -116,22 +168,23 @@ func (srv *Server) initAPI() {
 	// Wallet API Calls
 	if srv.wallet != nil {
 		router.GET("/wallet", srv.walletHandler)
-		router.POST("/wallet/033x", srv.wallet033xHandler)
-		router.GET("/wallet/address", srv.walletAddressHandler)
+		router.POST("/wallet/033x", requirePassword(srv.wallet033xHandler, password))
+		router.GET("/wallet/address", requirePassword(srv.walletAddressHandler, password))
 		router.GET("/wallet/addresses", srv.walletAddressesHandler)
-		router.GET("/wallet/backup", srv.walletBackupHandler)
-		router.POST("/wallet/init", srv.walletInitHandler)
-		router.POST("/wallet/lock", srv.walletLockHandler)
-		router.POST("/wallet/seed", srv.walletSeedHandler)
-		router.GET("/wallet/seeds", srv.walletSeedsHandler)
-		router.POST("/wallet/siacoins", srv.walletSiacoinsHandler)
-		router.POST("/wallet/siafunds", srv.walletSiafundsHandler)
-		router.POST("/wallet/siagkey", srv.walletSiagkeyHandler)
+		router.GET("/wallet/backup", requirePassword(srv.walletBackupHandler, password))
+		router.POST("/wallet/init", requirePassword(srv.walletInitHandler, password))
+		router.POST("/wallet/lock", requirePassword(srv.walletLockHandler, password))
+		router.POST("/wallet/seed", requirePassword(srv.walletSeedHandler, password))
+		router.GET("/wallet/seeds", requirePassword(srv.walletSeedsHandler, password))
+		router.POST("/wallet/siacoins", requirePassword(srv.walletSiacoinsHandler, password))
+		router.POST("/wallet/siafunds", requirePassword(srv.walletSiafundsHandler, password))
+		router.POST("/wallet/siagkey", requirePassword(srv.walletSiagkeyHandler, password))
```

Comment:
the usual way to do middleware is to wrap the router, not the individual handlers.

see below:
```go
 	// Apply UserAgent middleware and create HTTP server
 	uaRouter := requireUserAgent(router, srv.requiredUserAgent)
 	srv.apiServer = &http.Server{Handler: uaRouter}
```
this would become:
```go
 	// Apply UserAgent and authentication middleware and create HTTP server
 	uaRouter := requireUserAgent(router, srv.requiredUserAgent)
 	uaAuthRouter := requirePassword(uaRouter, password)
 	srv.apiServer = &http.Server{Handler: uaAuthRouter}
```
and `requirePassword` would need to be modified accordingly.
speaking of which, do we still need the UserAgent restriction if we have authentication?

I suppose the difference here is that we aren't applying auth to every endpoint, just the vast majority. We could use a whitelist, I suppose...or just require authentication for every call.